sklearn/utils/estimator_checks.py

```diff
@@ -816,7 +816,7 @@ def check_estimator_sparse_data(name, estimator_orig, strict_mode=True):
 
 
 @ignore_warnings(category=FutureWarning)
-def check_sample_weights_pandas_series(name, estimator_orig, strict_mode=True):
+def check_sample_weights_pandas_series(name, estimator_orig, api_only=False):
```

Comment:
My latest comment regarding several `mode` would be nice here. One mode would be the interoperability of array-like.
One might want to be API and array-like compliant while not be strict on the error message raised. In this case maybe `mode` is not a good name but I have in mind something like `model=["api", "interoperability"]`